### PR TITLE
Add support for the code_scanning_codeql_java_lombok feature flag

### PR DESCRIPTION
**Pull request by igfoo**
I think the `minimumVersion` could be older, but I don't think it's worth the time to work out exactly how old.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/feature-flags.ts [hunk 1]
```diff
@@ -58,6 +59,11 @@ export const featureConfig: Record<
   Feature,
   { envVar: string; minimumVersion: string | undefined; defaultValue: boolean }
 > = {
+  [Feature.CodeqlJavaLombokEnabled]: {
+    envVar: "CODEQL_JAVA_LOMBOK_ENABLED",
```
Comment (angelapwen) [inline src/feature-flags.ts:63]: For consistency with the others environment variable overrides for the feature flag! 
```suggestion
    envVar: "CODEQL_JAVA_LOMBOK",
```

**Review by angelapwen — DISMISSED** [review]
:shipit:

**Review by angelapwen — DISMISSED** [review]
That should do it 😸

**Review by angelapwen — APPROVED** [review]

File: src/init-action.ts [hunk 2]
```diff
@@ -342,6 +342,20 @@ async function run() {
       core.exportVariable("CODEQL_EXTRACTOR_JAVA_AGENT_DISABLE_KOTLIN", "true");
     }
 
+    if (await features.getValue(Feature.CodeqlJavaLombokEnabled, codeql)) {
+      logger.info("Enabling CodeQL Java Lombok support");
```
Comment (henrymercer) [inline src/init-action.ts:346]: Currently we'll print this even if the user isn't analyzing Java, which might be a bit confusing.  Could we run this later in the job, or alternatively change the message?

Comment (angelapwen) [reply src/init-action.ts:346]: Oh yes, good point. We could also check the feature flag only if one of the languages to be analyzed is Java, similar to what's done for the python dependencies check: https://github.com/github/codeql-action/blob/de6681ceb7967227f8760bfad7bde8540662f894/src/init-action.ts#L276

**Review by henrymercer — DISMISSED** [review]

**Review by henrymercer — APPROVED** [review]

**Pull request merged**